nengo/tests/test_connection.py

```diff
@@ -613,6 +613,45 @@ def test_slicing_function(Simulator, plt, seed):
     assert np.allclose(w, y, atol=0.1)
 
 
+def test_advanced_indexing(Simulator, plt, seed):
+    N = 100
+
+    model = nengo.Network()
+    with model:
+        u = nengo.Node([-1, 1])
+        a = nengo.Ensemble(n_neurons=N, radius=3, dimensions=1)
+        b = nengo.Ensemble(n_neurons=N, radius=3, dimensions=1)
+        c = nengo.Ensemble(n_neurons=N, radius=3, dimensions=2)
+        d = nengo.Ensemble(n_neurons=N, radius=3, dimensions=2)
+        nengo.Connection(u[[0, 1]], a[[0, 0]])
+        nengo.Connection(u[[1, 1]], b[[0, 0]])
+        nengo.Connection(u[[0, 1]], c[[0, 1]])
+        nengo.Connection(u[[1, 1]], d[[0, 1]])
+
+        a_probe = nengo.Probe(a, synapse=0.01)
+        b_probe = nengo.Probe(b, synapse=0.01)
+        c_probe = nengo.Probe(c, synapse=0.01)
+        d_probe = nengo.Probe(d, synapse=0.01)
+
+    with nengo.Simulator(model) as sim:
+        sim.run(0.4)
+
+    t = sim.trange()
+    a_data = sim.data[a_probe]
+    b_data = sim.data[b_probe]
+    c_data = sim.data[c_probe]
+    d_data = sim.data[d_probe]
+
+    plt.plot(t, a_data)
+    assert np.allclose(a_data[100:], [0], atol=0.3)
+    plt.plot(t, b_data)
+    assert np.allclose(b_data[100:], [2], atol=0.3)
+    plt.plot(t, d_data)
+    assert np.allclose(c_data[100:], [-1,1], atol=0.3)
+    plt.plot(t, d_data)
+    assert np.allclose(d_data[100:], [1,1], atol=0.3)
```

Comment:
This test is great, thanks!

We mentioned in the dev meeting that it would also be nice to support boolean arrays for indices. Have you tried testing with boolean indices? If not, it might be worth trying it out to see if it works... if it doesn't, it makes sense to implement it in another PR, but if it just works then it'd be good to add another test for that here!

Reply:
I tried this and it seems to work. I'll add a test for it.